substrate/frame/try-runtime/src/lib.rs

```diff
@@ -18,36 +18,9 @@
 //! Supporting types for try-runtime, testing and dry-running commands.
 
 #![cfg_attr(not(feature = "std"), no_std)]
-#![cfg(feature = "try-runtime")]
```

Comment:
It makes no sense that this crate has a `try-runtime` feature.. This crate should only be activated if you use `try-runtime`.

Reply:
Then it would enable the `try-runtime` feature of all dependencies by default?  
I dont remember if this can cause issues with feature unification, but maybe it works since the crate would disabled otherwise.

Reply:
Okay actually this will make `--workspace` not compile anymore, since it enables all crates, you can check it with `cargo tree --workspace` and see that `frame-system-benchmarking` is enabled as top-level @bkchr.

Reply:
Hmm yeah okay... :seal: 

Maybe having the `try_runtime` types in `frame-support` not behind a feature would be a better solution.

Reply:
We have some traits with feature gated functions. Maybe that is an anti-pattern in itself... should probably be an extension trait that is only selectively required.

Reply:
I mean the functions is fine, I would probably just not hide the traits etc in `try_runtime.rs`. But in the end this is all discussable and we can keep it like it is done right now.

Reply:
Generally just don't like these `inner.rs` files.